File: Day 5/refactoredTasks/exponent/exponent.js
// X to the power of Y

// ? Our original implementation
function exponentOriginal(x, y) {
  let product = 1;

  for (let i = 0; i < y; i++) {
    product = product * x;
  }

  return product;
}

// ? Refactored
function exponent(x, y) {
  // ? '**' is the arithmetic operator for 'power of'
  return x ** y;
}

File: Day 5/refactoredTasks/exponent/exponent.py
# X to the power of Y

# ? Our original implementation
def exponentOriginal(x, y):
    product = 1

    for i in range(0, y):
        product = product * x

    return product


# ? Refactored
def exponent(x, y):
    return x ** y  # ? '**' is the arithmetic operator for 'power of'
